packages/smooth_app/lib/background/background_task_language_refresh.dart

```diff
@@ -0,0 +1,136 @@
+import 'package:flutter/painting.dart';
+import 'package:flutter_gen/gen_l10n/app_localizations.dart';
+import 'package:openfoodfacts/openfoodfacts.dart';
+import 'package:smooth_app/background/background_task.dart';
+import 'package:smooth_app/background/operation_type.dart';
+import 'package:smooth_app/database/dao_product.dart';
+import 'package:smooth_app/database/local_database.dart';
+import 'package:smooth_app/query/product_query.dart';
+
+/// Background task about downloading products to translate.
+class BackgroundTaskLanguageRefresh extends BackgroundTask {
+  BackgroundTaskLanguageRefresh._({
+    required super.processName,
+    required super.uniqueId,
+    required super.stamp,
+    required this.excludeBarcodes,
+  });
+
+  BackgroundTaskLanguageRefresh.fromJson(Map<String, dynamic> json)
+      : excludeBarcodes = _getStringList(json, _jsonTagExcludeBarcodes),
+        super.fromJson(json);
+
+  static List<String> _getStringList(
+      final Map<String, dynamic> json, final String tag) {
+    final List<dynamic> dynamicList =
+        json[_jsonTagExcludeBarcodes] as List<dynamic>;
+    final List<String> result = <String>[];
+    for (final dynamic item in dynamicList) {
+      result.add(item.toString());
+    }
+    return result;
+  }
+
+  final List<String> excludeBarcodes;
+
+  static const String _jsonTagExcludeBarcodes = 'excludeBarcodes';
+
+  @override
+  Map<String, dynamic> toJson() {
+    final Map<String, dynamic> result = super.toJson();
+    result[_jsonTagExcludeBarcodes] = excludeBarcodes;
+    return result;
+  }
+
+  static const OperationType _operationType = OperationType.languageRefresh;
+
+  static Future<void> addTask(
+    final LocalDatabase localDatabase, {
+    final List<String> excludeBarcodes = const <String>[],
+  }) async {
+    final String uniqueId = await _operationType.getNewKey(
+      localDatabase,
+    );
+    final BackgroundTask task = _getNewTask(
+      uniqueId,
+      excludeBarcodes,
+    );
+    await task.addToManager(localDatabase);
+  }
+
+  @override
+  (String, AlignmentGeometry)? getFloatingMessage(
+          final AppLocalizations appLocalizations) =>
+      null;
+
+  static BackgroundTask _getNewTask(
+    final String uniqueId,
+    final List<String> excludeBarcodes,
+  ) =>
+      BackgroundTaskLanguageRefresh._(
+        processName: _operationType.processName,
+        uniqueId: uniqueId,
+        stamp: ';languageRefresh',
+        excludeBarcodes: excludeBarcodes,
+      );
+
+  @override
+  Future<void> preExecute(final LocalDatabase localDatabase) async {}
+
+  @override
+  bool get hasImmediateNextTask => true;
```

Comment:
It's not always the case. I'm again doing a mobile review so I can't double-check the implications but I guess it's not causing any problems, is it?

Reply:
```dart
bool get hasImmediateNextTask => true;
```

@M123-dev When we page our background tasks, task A will deal with only say 10 products and at the end will say "I know there are more products to deal with and I create task B". In that case we need the background manager to immediately restart, to deal with task B.
In case we have no more products to deal with, no additional task will be created and the manager will see an empty task queue, and do nothing.

As opposed to standard one-shot tasks (e.g. product detail change), that are not somehow chained to other tasks to be run immediately.

Reply:
> I'm again doing a mobile review so I can't double-check the implications

Many thanks @M123-dev, that's not the easiest PR to review on mobile ;)